clearly-defined/src/main/kotlin/Extensions.kt

```diff
@@ -17,12 +17,34 @@
  * License-Filename: LICENSE
  */
 
+@file:Suppress("MatchingDeclarationName")
+
 package org.ossreviewtoolkit.clearlydefined
 
 import okhttp3.RequestBody
 
 import okio.Buffer
 
+/**
+ * The status of meta-data harvesting of the various tools.
+ */
+enum class HarvestStatus {
+    NOT_HARVESTED,
+    PARTIALLY_HARVESTED,
+    HARVESTED
+}
+
+/**
+ * Return the harvest status of a described component.
+ */
+fun ClearlyDefinedService.Defined.getHarvestStatus() =
+    // See https://github.com/clearlydefined/website/blob/de42d2c/src/components/Navigation/Ui/HarvestIndicator.js#L8.
+    when {
+        described.tools == null -> HarvestStatus.NOT_HARVESTED
+        described.tools.size > 2 -> HarvestStatus.HARVESTED
+        else -> HarvestStatus.PARTIALLY_HARVESTED
+    }
```

Comment:
Why don't you add this code to `ClearlyDefinedService.kt`?

Reply:
Basically, for the same reason @fviernau mentioned [here](https://github.com/oss-review-toolkit/ort/pull/2578#discussion_r420803233): To completely separate the data model from UI-related functionality. But I agree this is an edge case: While in the original JavaScript version this code was used to display a coarse UI progress indicator, it's not really a UI function that creates a display representation.

Another reason was that `ClearlyDefinedService.kt` already is quite big, and I was considering to split it up somehow anyway, so I felt like starting with that as I already had `Extensions.kt`.

But I do not care strongly. I'll happily move it if you feel it should be moved.

Reply:
To me it would make more sense to add the enum and function to the service class, I'm just concerned that it's bad for the overview if part of the functionality is in the service class and another part in the extensions file.
One difference in @fviernau's PR is that there the extension function is only used in a different module than the one that contains the model class.